### ISSUE TITLE
az postgres flexible-server execute cannot read UTF-8 with BOM SQL files

### ISSUE THREAD

**Issue by julian-code** [Azure/azure-cli#23236]
**Related command**
`az postgres flexible-server execute`

**Describe the bug**
Using the dotnet-ef tool https://docs.microsoft.com/en-us/ef/core/cli/dotnet#dotnet-ef-migrations-script it generates a UTF-8 BOM SQL file. When send to `az postgres flexible-server execute` via `-f` it fails with following error:
```
2022-07-18T12:19:53.7465981Z ERROR: syntax error at or near "﻿CREATE"
2022-07-18T12:19:53.7466937Z LINE 1: ﻿CREATE TABLE IF NOT EXISTS "__EFMigrationsHistory" (
2022-07-18T12:19:53.7467668Z         ^
2022-07-18T12:19:53.7467916Z 
```

**To Reproduce**
1. Create a SQL file with UTF-8 with BOM encoding
2. Send it via `az postgres flexible-server execute`

**Expected behavior**
It should be possible to read UTF-8 with BOM encoded SQL files.

**Environment summary**
`az --version` dump
```
azure-cli                         2.38.0

core                              2.38.0
telemetry                          1.0.6

Extensions:
azure-devops                      0.25.0
rdbms-connect                      1.0.2

Dependencies:
msal                            1.18.0b1
azure-mgmt-resource             21.1.0b1
```

**Additional context**
The flexible server is completely as default. Maybe it is a postgres setting, or the rdbms-connect tool which is required to run the `execute` command that can't handle UTF-8 with BOM.

**Comment by SaurabhSharma-MSFT** [comment]
@julian-code Engg. team working on finding the root cause of it. I will update you as soon as I hear back from them.

**Comment by alanenriqueo** [comment]
@julian-code The bug was fixed in rdbms-connect 1.0.3, can you please check?

**Comment by SaurabhSharma-MSFT** [comment]
@julian-code Please let me know if you are seeing any issues.
